src/core/src/signature.rs

```diff
@@ -20,138 +20,279 @@ use crate::sketch::Sketch;
 use crate::Error;
 use crate::HashIntoType;
 
-pub trait SigsTrait {
-    fn size(&self) -> usize;
-    fn to_vec(&self) -> Vec<u64>;
-    fn ksize(&self) -> usize;
-    fn check_compatible(&self, other: &Self) -> Result<(), Error>;
-    fn seed(&self) -> u64;
+// Iterator for converting sequence to hashes
+pub struct SeqToHashes {
+    sequence: Vec<u8>,
+    kmer_index: usize,
+    k_size: usize,
+    max_index: usize,
+    force: bool,
+    is_protein: bool,
+    hash_function: HashFunctions,
+    seed: u64,
+    _hashes_buffer: Vec<u64>,
+
+    _dna_configured: bool,
+    _dna_rc: Vec<u8>,
+    _dna_ksize: usize,
+    _dna_len: usize,
+    _dna_last_position_check: usize,
+
+    _prot_configured: bool,
+    _aa_seq: Vec<u8>,
```

Comment:
any strong reason for these fields all start with `_`? They are all private by default, so `_` doesn't add much info...

Reply:
No, just to easy-discriminate the iterator variables.

Reply:
Resolved in [`9bed5d9` (#1653)](https://github.com/sourmash-bio/sourmash/pull/1653/commits/9bed5d9cf448639647c159562cb585b12b505f6b)